src/main/java/seedu/address/ui/control/PopupAddInternship.java

```diff
@@ -0,0 +1,328 @@
+
+    @FXML
+    private TextField companyName;
+    @FXML
+    private TextField jobTitle;
+    @FXML
+    private TextField place;
+    @FXML
+    private TextField salary;
+    @FXML
+    private TextField rating;
+    @FXML
+    private VBox qualificationVBox;
+    @FXML
+    private TextField qualification;
+    @FXML
+    private VBox programmingLanguageVBox;
+    @FXML
+    private TextField programmingLanguage;
+    @FXML
+    private VBox reviewVBox;
+    @FXML
+    private TextField review;
+    @FXML
+    private VBox noteVBox;
+    @FXML
+    private TextField note;
+    @FXML
+    private VBox reflectionVBox;
+    @FXML
+    private TextField reflection;
+
+    /**
+     * Creates a new popup add internship.
+     */
+    public PopupAddInternship(Stage root) {
+        super(FXML, root);
+    }
+    /**
+     * Creates a new popup add internship.
+     */
+    public PopupAddInternship(MainWindow mainWindow) {
+        this(new Stage());
+        this.mainWindow = mainWindow;
+    }
+
+    /**
+     * Shows the help window.
+     * @throws IllegalStateException
+     *     <ul>
+     *         <li>
+     *             if this method is called on a thread other than the JavaFX Application Thread.
+     *         </li>
+     *         <li>
+     *             if this method is called during animation or layout processing.
+     *         </li>
+     *         <li>
+     *             if this method is called on the primary stage.
+     *         </li>
+     *         <li>
+     *             if {@code dialogStage} is already showing.
+     *         </li>
+     *     </ul>
+     */
+    public void show() {
+        logger.fine("Showing add internship popup");
+        companyName.setText("");
+        jobTitle.setText("");
+        place.setText("");
+        salary.setText("");
+        rating.setText("");
+
+        qualification.setText("");
+        int qVBoxSize = qualificationVBox.getChildren().size();
+        if (qVBoxSize > 2) {
+            qualificationVBox.getChildren().remove(2, qVBoxSize);
+        }
+
+        programmingLanguage.setText("");
+        int plVBoxSize = programmingLanguageVBox.getChildren().size();
+        if (plVBoxSize > 2) {
+            programmingLanguageVBox.getChildren().remove(2, plVBoxSize);
+        }
+
+        review.setText("");
+        int rVBoxSize = reviewVBox.getChildren().size();
+        if (rVBoxSize > 2) {
+            reviewVBox.getChildren().remove(2, rVBoxSize);
+        }
+
+        note.setText("");
+        int nVBoxSize = noteVBox.getChildren().size();
+        if (nVBoxSize > 2) {
+            noteVBox.getChildren().remove(2, nVBoxSize);
+        }
+
+        reflection.setText("");
+        int rfVBoxSize = reflectionVBox.getChildren().size();
+        if (rfVBoxSize > 2) {
+            reflectionVBox.getChildren().remove(2, rfVBoxSize);
+        }
+
+        getRoot().show();
+        getRoot().centerOnScreen();
+    }
```

Comment:
This method seems too long to me. Since there has no method parameter, would it be better to extract them out as helper functions.

Reply:
The  setting text and removing children can be extracted out but it will still be quite long because the number of attributes in Add Command.